lib/codecs/src/encoding/format/gelf.rs

```diff
@@ -165,6 +167,19 @@ fn coerce_field_names_and_values(
                     if !(value.is_timestamp() || value.is_integer()) {
                         err_invalid_type(field, "timestamp or integer", value.kind_str())?;
                     }
+
+                    // timestamp to gelf compatible f64 (milliseconds as decimal)
+                    if let Value::Timestamp(ts) = value {
+                        if ts.timestamp_subsec_millis() > 0 {
+                            *value = Value::Float(
+                                NotNan::new(ts.timestamp_millis() as f64 / 1000.0).unwrap(),
```

Comment:
Not sure it would save that much but it strikes me that you could get the timestamp in milliseconds once and do the check with arithmetic operations on that without a separate function call.

Non-blocking

Reply:
Sounds absolutely reasonable to me. Thx for pointing that out.